cmd/agent/app/integrations.go

```diff
@@ -0,0 +1,269 @@
+	tufCmd.PersistentFlags().MarkHidden("cmd-flags")
+	tufCmd.PersistentFlags().MarkHidden("idx-flags")
+	tufCmd.PersistentFlags().MarkHidden("use-sys-python")
+}
+
+var tufCmd = &cobra.Command{
+	Use:   "integration [command]",
+	Short: "Datadog integration manager (ALPHA feature)",
+	Long:  ``,
+}
+
+var installCmd = &cobra.Command{
+	Use:   "install [package]",
+	Short: "Install Datadog integration core/extra packages",
+	Long:  ``,
+	RunE:  installTuf,
+}
+
+var removeCmd = &cobra.Command{
+	Use:   "remove [package]",
+	Short: "Remove Datadog integration core/extra packages",
+	Long:  ``,
+	RunE:  removeTuf,
+}
+
+var searchCmd = &cobra.Command{
+	Use:   "search [package]",
+	Short: "Search Datadog integration core/extra packages",
+	Long:  ``,
+	RunE:  searchTuf,
+}
+
+func getTufConfigPath() string {
+	here, _ := executable.Folder()
+	return filepath.Join(here, relTufConfigFilePath)
+}
+
+func getCommandPython() (string, error) {
+	if useSysPython {
+		return pythonBin, nil
+	}
+
+	here, _ := executable.Folder()
+	pyPath := filepath.Join(here, relPyPath)
+
+	if _, err := os.Stat(pyPath); err != nil {
+		if os.IsNotExist(err) {
+			return pyPath, errors.New("unable to find pip executable")
+		}
+	}
+
+	return pyPath, nil
+}
+
+func getConstraintsFilePath() (string, error) {
+	here, _ := executable.Folder()
+	cPath := filepath.Join(here, relConstraintsPath)
+
+	if _, err := os.Stat(cPath); err != nil {
+		if os.IsNotExist(err) {
+			return cPath, err
+		}
+	}
+
+	return cPath, nil
+}
+
+func getTUFConfigFilePath() (string, error) {
+	if _, err := os.Stat(tufConfig); err != nil {
+		if os.IsNotExist(err) {
+			return tufConfig, err
+		}
+	}
+
+	return tufConfig, nil
+}
+
+func tuf(args []string) error {
+	if !allowRoot && !authorizedUser() {
+		return errors.New("Please use this tool as the agent-running user")
+	}
+
+	if flagNoColor {
+		color.NoColor = true
+	}
+
+	err := common.SetupConfig(confFilePath)
+	if err != nil {
+		fmt.Printf("Cannot setup config, exiting: %v\n", err)
+		return err
+	}
+
+	pipPath, err := getCommandPython()
+	if err != nil {
+		return err
+	}
+	tufPath, err := getTUFConfigFilePath()
+	if err != nil && !withoutTuf {
+		return err
+	}
+
+	cmd := args[0]
+	implicitFlags := args[1:]
+	args = append([]string{"-mpip"}, cmd)
+
+	// Add pip power-user flags
+	// cmd-flags go before the actual command
+	cmdFlags, err := tufCmd.Flags().GetStringSlice("cmd-flags")
+	if err == nil {
+		args = append(args, cmdFlags...)
+	}
+	args = append(args, implicitFlags...)
+
+	// Proxy support
+	proxies, err := config.GetProxies()
+	if err == nil && proxies != nil {
+		proxyFlags := fmt.Sprintf("--proxy=%s", proxies.HTTPS)
+		args = append(args, proxyFlags)
+	}
+
+	// idx-flags go after the command and implicit flags
+	idxFlags, err := tufCmd.Flags().GetStringSlice("idx-flags")
+	if err == nil {
+		args = append(args, idxFlags...)
+	}
+
+	tufCmd := exec.Command(pipPath, args...)
+
+	if !withoutTuf {
+		tufCmd.Env = append(os.Environ(),
```

Comment:
I think by default Exec will set the current ENV as the environment of the command. Are we worried about leaking information to TUF ? Like api_key, proxy passwords ...

Reply:
This is a great question. Strictly speaking we do care. That said, if a person can run the command you're basically already toast because if you can run commands as the `dd-agent` user you can already inspect the environment of the running agent, so there's not much additional damage that can be done. Also, to actually expose the secrets they should be able to modify the underlying pip to access `os.environ`....

Reply:
Correct. The default pip+TUF code does not access any other environment variable other than the TUF configuration file.

Reply:
One thing we can do here is to make a blank copy of the environment, and pass that instead.